src/clarity-angular/datagrid/render/column-resizer.ts

```diff
@@ -0,0 +1,190 @@
+
+        mouseDownListener = this.renderer.listen(this.draggerEl, "mousedown", () => {
+            this.onDragStart();
+
+            mouseMoveListener = this.renderer.listenGlobal("document", "mousemove", (mouseMoveEvent: any) => {
+                this.onDragMove(mouseMoveEvent);
+            });
+
+            mouseUpListener = this.renderer.listenGlobal("document", "mouseup", () => {
+
+                //Unsubscribing from mouseMoveListener
+                mouseMoveListener();
+                this.onDragEnd();
+
+                //Unsubscribing from itself
+                mouseUpListener();
+            });
+        });
+
+        touchStartListener = this.renderer.listen(this.draggerEl, "touchstart", () => {
+            this.onDragStart();
+
+            touchMoveListener = this.renderer.listenGlobal("document", "touchmove", (touchMoveEvent: any) => {
+                this.onDragMove(touchMoveEvent);
+            });
+
+            touchEndListener = this.renderer.listenGlobal("document", "touchend", () => {
+
+                //Unsubscribing from mouseMoveListener
+                touchMoveListener();
+                this.onDragEnd();
+
+                //Unsubscribing from itself
+                touchEndListener();
+            });
+        });
+
+        //returning mouseDownListener and touchStartListener so they can be unsubscribed.
+        return [mouseDownListener, touchStartListener];
+    }
+
+    onDragStart(): void {
+
+        this.renderer.setElementStyle(this.draggerTracker, "display", "block");
+
+        this.dragDistancePositionX = 0;
+        this.columnRectWidth = this.domAdapter.clientRectWidth(this.columnEl);
+        this.pageStartPositionX = this.domAdapter.clientRectRight(this.columnEl);
+
+    }
+
+    onDragMove(moveEvent: any): void {
+        let pageMovePosition = moveEvent.pageX || moveEvent.changedTouches[0].pageX;
+
+        this.dragDistancePositionX = this.getPositionWithinMax(pageMovePosition - this.pageStartPositionX);
+
+        this.renderer.setElementStyle(this.draggerTracker, "right", -1 * this.dragDistancePositionX + "px");
+        this.renderer.setElementStyle(document.body, "cursor", "col-resize");
+    }
+
+    onDragEnd(): void {
+        // on drag end, the right position should get back to its original position,
+        // which is right: -12px; in .datagrid-draggable-tracker
+        this.renderer.setElementStyle(this.draggerTracker, "right", "0px");
+        this.renderer.setElementStyle(this.draggerTracker, "display", "none");
+        this.renderer.setElementStyle(document.body, "cursor", "auto");
+
+        if (this.dragDistancePositionX) {
+
+            this.columnResizeBy = this.dragDistancePositionX;
+
+            this.resizeEmitter.emit({
+                change: this.columnResizeBy,
+                newWidth: this.columnRectWidth + this.columnResizeBy,
+                minWidth: this.columnMinWidth
+            });
+
+            this.organizer.resize();
```

Comment:
I don't know what happened, but this architecture changed since the last time I saw it. Or I misread last time, but either way I'm absolutely sure I explained how I wanted the resizer to update the organizer's strict widths directly, as opposed to having the various directives being so tightly coupled.

The problem is that this would be a pretty large change, and I don't want to postpone this anymore if it's working fine right now. But I'm definitely not happy with that component communication, it makes the various directives depend far too much on one another, which completely defeats the purpose of having multiple directives on the same selector.